### DISCUSSION TITLE
"Fetch Tables" from MySQL deletes all existing changes / customisations. Can't fetch new tables

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#5539] [Help]
Scenario (real)
I am building a Clinic Management System for my local Children's Clinic, not an internal app. This will be their main tool.
When finished, it will be available for anyone who wishes to use it.

As i am progressing, i am adding new tables to my MySQL DB
I am not adding the tables from inside BB, as some options aren't yet supported ( ENUM )
Now I want to fetch this new table to BB, but i don't want to override all the changes I have already made to existing tables inside BB, like formula columns or manually entered ENUM options.

An option to select which tables to refresh from the DB and which to be left untouched would be most useful in this case.

**Comment by poirazis** [comment]
... and i am stuck! I can't bring in new tables i created without loosing the changes / customisations to the existing ones.

**Reply by melohagan** [reply]
Hey @poirazis Unfortunately this can't be done at the moment. There is a feature request out here: https://github.com/Budibase/budibase/issues/4651

**Reply by giacomouberti** [reply]
Hi @melohagan, the functionalities requested on [#4651] would be excellent in allowing fetching selected tables but if one of the tables that need to be fetched **has any formula column that would be wiped anyway** is there any chance that this issue could be fixed?

**Comment by poirazis** [comment]
Hey @melohagan 
I have temporarily worked my way around it by manually editing the CoundDB table, until synchronisation is supported.

Have a blessed day.

**Reply by melohagan** [reply]
Thanks, you too!

**Reply by poirazis** [reply]
P.S maybe give a warning that clicking Fetch will remove previous changes made so future users know

**Reply by mike12345567** [reply]
Hi @poirazis - just to check which changes are being wiped? For the most part Budibase should conserve changes which you've made in the UI (e.g. changing a column from text to options type) if there is a particular type in a particularly database that isn't being conserved it'd be good to know so that it can be fixed.

Or is this specifically with extra added columns? There could be an issue with extra additional columns being removed which needs to be fixed (all column types should be added to the underlying DB - apart from formulas).

**Reply by poirazis** [reply]
Hey @mike12345567 

Formula Columns get wiped. 
Table Display Columns gets wiped if changed.
Columns created from within BB don't keep their config. For example, you have a table patients, from within BB you added a column of type Test, when you fetch this new column will come back as a LongFormText.

Another case is a field gets renamed in the underlying DB which is used in the Definition of Existing Relationship ( which would mess up the Relationship )

I guess one way would be, to be able to select which tables to fetch. That way you could pull in new tables, of force refresh an existing that had to be changed. In the second case, i think there should be a "validateRelationships" which would mark as erroneous the ones that needs attention.

Another useful option would be to give the user the ability to remove a table (unlink from BB).

Refresh / Unlink could possibly be options of the table menu.
Link Tables ( Current Fetch All ) could reside where it is currently, giving a list of all NON linked tables and letting the user select ( multiple ).

in my humble view, if db management ( create tables / columns / rels ) for MySQL is to be provided, it would eventually need to be complete with all the options and settings. otherwise it would be very difficult to keep in sync over time.

Just some thought! :)

**Reply by mike12345567** [reply]
Hi @poirazis - i'll take a look at some of those initial issues around columns created in Budibase changing types and formulas disappearing those shouldn't happen.

In terms of changing the name of a field outside of Budibase, in that situation it wouldn't really be possible for Budibase to be aware that they were the same column and pair them up - I can see the problem there.

You're probably right that the best solution would be to have a way to selectively update tables to avoid issues with tables that have already been defined - as @melohagan said we have an issue for that and it will be worked on in due course.

Thanks for the input!

**Reply by poirazis** [reply]
Happy to help !
